auth/services/selfsigned/store.go

```diff
@@ -65,3 +74,30 @@ func (s *memorySessionStore) Delete(sessionID string) {
 	defer s.lock.Unlock()
 	delete(s.sessions, sessionID)
 }
+
+func (s *memorySessionStore) Start(ctx context.Context) {
+	done := ctx.Done()
+
+	go func() {
+		timer := time.NewTimer(s.expiryInterval)
+		for {
+			select {
+			case <-done:
+				return
+			case <-timer.C:
+				s.evict()
+			}
+		}
+	}()
+}
+
+func (s *memorySessionStore) evict() {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	for k, v := range s.sessions {
+		if v.ExpiresAt.Before(time.Now()) {
```

Comment:
Now certain states can't be retrieved anymore by the REST client (e.g. expired), leading them to be functionally dead code, maybe evict it a minute or so after expiry instead of straight away.

Reply:
I'll add a 10 minute delay